store/errors.go

```diff
@@ -51,3 +51,12 @@ type ErrDownload struct {
 func (e *ErrDownload) Error() string {
 	return fmt.Sprintf("received an unexpected http response code (%v) when trying to download %s", e.Code, e.URL)
 }
+
+type HTTPError struct {
```

Comment:
Aha, so here it is. Can we please avoid that sort of logic? This is our API.. we should return an error message that can generally be consumed by our own logic an produce sane error messages that give a hint of what is wrong. Compare this error type with the above one for ErrDownload (which is misnamed should be DownloadError), for example.